bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerApiFactory.java

```diff
@@ -58,12 +58,12 @@ public OpenSprinklerApi getHttpApi(OpenSprinklerHttpInterfaceConfig config)
 
         OpenSprinklerApi lowestSupportedApi = new OpenSprinklerHttpApiV100(this.httpClient, config);
         try {
+            lowestSupportedApi.refresh();
```

Comment:
the idea of the API class is to abstract away how the data is fetched. I would remove the call to refresh here and move it either to the constructor of OpenSprinklerHttpApiV100, or make it implicit when calling a getter function and the data is not loaded already.

Reply:
Thank you for the feedback, OK I can look to move it to the constructor for the initial refresh. To give background on why this change was made.

This solved two issues:

1. Each channel on each poll would fetch the data again, so with 200 stations (project allows up to 200 stations) and 4 channels on each station it would be sending >800 GET requests each poll. Reply needed to be requested once and then re-used for each channel.
2. When an exception occurred it was sending it to the DEVICE and not the BRIDGE. The online and offline was made more simple by using the frameworks ability to take child things offline when the bridge goes offline.

The Polling task fires the refresh() method so only 1 GET request is sent and the BRIDGE can catch the exceptions for online and offline control.

Reply:
DONE: I moved it to the getter function as it was already setup to handle the THROWS.